# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../cifs-utils/cifs-utils-6.0-0001-upstream-21b5bdc.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From 21b5bdcb354bc113473347af5e4995fae2285b58 Mon Sep 17 00:00:00 2001
From: Peng Haitao <penght@cn.fujitsu.com>
Date: Fri, 12 Apr 2013 16:58:49 +0800
Subject: [PATCH] setcifsacl.c: fix a bug of goto setcifsacl_facenum_ret

setcifsacl_facenum_ret: is called only if attrlen is equal to -1.

Signed-off-by: Peng Haitao <penght@cn.fujitsu.com>
---
 setcifsacl.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/setcifsacl.c b/setcifsacl.c
index 67dd29c..4b38459 100644
--- a/setcifsacl.c
+++ b/setcifsacl.c
@@ -822,9 +822,10 @@ cifsacl:
 		goto setcifsacl_action_ret;
 
 	attrlen = setxattr(filename, ATTRNAME, ntsdptr, bufsize, 0);
-	if (attrlen == -1)
+	if (attrlen == -1) {
 		printf("%s: setxattr error: %s\n", __func__, strerror(errno));
-	goto setcifsacl_facenum_ret;
+		goto setcifsacl_facenum_ret;
+	}
 
 	exit_plugin(plugin_handle);
 	return 0;
-- 
1.7.0.4

